server/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -216,6 +240,22 @@ private GetResult innerGet(
         }
     }
 
+    private Engine.GetResult getFromIndexShard(
```

Comment:
I don't see much value in adding this method; I would prefer something like:
```java
var engineGet = new Engine.Get(realtime, realtime, id).version(version)
            .versionType(versionType)
            .setIfSeqNo(ifSeqNo)
            .setIfPrimaryTerm(ifPrimaryTerm);

try (Engine.GetResult get = translogOnly ? indexShard.getFromTranslog(engineGet) : indexShard.get(engineGet)) {
...
}
```

Reply:
Somehow I thought that might be more readable! But I've changed it now to your suggestion.